src/lib/InstantSearch.ts

```diff
@@ -453,8 +462,10 @@ See: https://www.algolia.com/doc/guides/building-search-ui/going-further/backend
   }
 
   public onStateChange = () => {
-    // @TODO: Provide `nextUiState` to all middlewares (eg. routing)
-    // const nextUiState = this.mainIndex.getWidgetState({});
+    const nextUiState = this.mainIndex.getWidgetState({});
```

Comment:
This is computed all the time while it should only be if `routing` is defined. It should have been part of the `if` branch below.